### PR TITLE
SQL changes on the MRI side. 

### PR DESCRIPTION
**Pull request by cmadjar**
1) Dropped tarchive_find_new_uploads table as it is no longer used in LORIS. However, the drop statement is commented in case older projects have data in that table. Make sure to check that table is empty before uncommenting it.

2) Added a foreign key constraint to the two following tables so that they are linked to the tarchive table using the TarchiveID field. Affected tables are:
- MRICandidateErrors
- mri_violations_log 

Note: in order to add the foreign key constraints, the creation of the tarchive tables had to be moved before the creation of the mri violation tables.

### CODE REVIEW

**Comment by cmadjar** [comment]
@MounaSafiHarab, I assigned this to you so that you can review it. It is for 19.0 release. Thanks!

File: SQL/Archive/19.0/2017-11-29-Drop_tarchive_find_new_uploads.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- UNCOMMENT THE FOLLOWING STATEMENT IF tarchive_find_new_uploads IS EMPTY
+-- THIS TABLE IS NOT USED ANYMORE AND WILL NEVER BE FILLED
+
+-- DROP TABLE IF EXISTS tarchive_find_new_uploads;
```
Comment (driusan) [inline SQL/Archive/19.0/2017-11-29-Drop_tarchive_find_new_uploads.sql]: I don't think this should be commented out.. there's no real data in the table, and no one uses it. It's better to not have hidden things that people need to know they need to do separately when upgrading unless there's a strong justification for it.

**Comment by cmadjar** [comment]
Done :). Thanks!

**Comment by cmadjar** [comment]
Hi @nicolasbrossard, Do you think you could test this PR? Thanks!

**Comment by cmadjar** [comment]
Hi Shen,
I have updated the automated test for the imaging browser due to the new foreign keys. If you could test it that would be great!!

Cécile

**Comment by nicolasbrossard** [comment]
Inserting a scan on my sandbox works fine with your changes. Tagged as 'Passed Manual Tests'

**Comment by MounaSafiHarab** [comment]
after this change, projects that have deleted specific files from the files table can no longer do this without deleting references to it in other tables; such as files_qcstatus; which is great.
But some projects might fail running this patch due to the fact that there might be some references to those fileIDs in other tables due to the absence of this exact implementation proposed here (CCNA is one such example!).
@cmadjar @driusan 
I think we should somehow document this to projects if the upgrade process fails for them?

**Comment by cmadjar** [comment]
@driusan I am working on documenting this. What would be the best way to do it? 

Here is the problematic: before running the patch associated to this PR, users should make sure that the following queries do not return anything. If it does, it means that someone manually deleted an entry from one table but not from the other. They would need to check "those entries with no parents" and most probably delete them before the patch can run successfully. I am not sure we should give directly the delete statements?

What do you think?


## Queries to run and that should return no rows. If it does, need to clean up the entries with no parent key.
```
-- check that no TarchiveID are present in MRICandidateErrors and not in tarchive table
SELECT MRICandidateErrors.TarchiveID AS MRICandidateErrors_TarchiveID, tarchive.TarchiveID AS tarchive_TarchiveID, ArchiveLocation AS tarchive_ArchiveLocation FROM MRICandidateErrors LEFT JOIN tarchive USING (TarchiveID) WHERE tarchive.TarchiveID IS NULL AND MRICandidateErrors.TarchiveID IS NOT NULL;

-- check that no TarchiveID are present in mri_violations_log and not in tarchive table
SELECT mri_violations_log.TarchiveID AS mri_violations_log_TarchiveID, tarchive.TarchiveID AS tarchive_TarchiveID, ArchiveLocation AS tarchive_ArchiveLocation FROM mri_violations_log LEFT JOIN tarchive USING (TarchiveID) WHERE tarchive.TarchiveID IS NULL AND mri_violations_log.TarchiveID IS NOT NULL;

-- check that no TarchiveSource/TarchiveID are present in files and not in tarchive table
SELECT files.TarchiveSource AS files_TarchiveSource, tarchive.TarchiveID AS tarchive_TarchiveID, ArchiveLocation AS tarchive_ArchiveLocation FROM files LEFT JOIN tarchive ON (files.TarchiveSource=tarchive.TarchiveID) WHERE tarchive.TarchiveID IS NULL AND files.TarchiveSource IS NOT NULL;

-- check that no FileID are present in files_qcstatus and not in files table
SELECT files_qcstatus.FileID AS files_qcstatus_FileID, files.FileID AS f_FileID, File AS files_File FROM files_qcstatus LEFT JOIN files USING (FileID) WHERE files.FileID IS NULL AND files_qcstatus.FileID IS NOT NULL;

-- check that no TarchiveID are present in mri_upload and not in tarchive table
SELECT mri_upload.TarchiveID AS mri_upload_TarchiveID, tarchive.TarchiveID AS tarchive_TarchiveID, ArchiveLocation AS tarchive_ArchiveLocation FROM mri_upload LEFT JOIN tarchive USING (TarchiveID) WHERE tarchive.TarchiveID IS NULL AND mri_upload.TarchiveID IS NOT NULL;

-- check that no SessionID are present in mri_upload and not in session table
SELECT mri_upload.SessionID AS mri_upload_SessionID, session.ID AS session_ID, CandID AS session_CandID, Visit_label AS session_visit FROM mri_upload LEFT JOIN session ON (session.ID=mri_upload.SessionID) WHERE session.ID IS NULL AND mri_upload.SessionID IS NOT NULL;

-- check that no Scan_type are present in mri_protocol_checks and not in mri_scan_type table
SELECT mri_protocol_checks.Scan_type AS mri_protocol_checks_Scan_type, mri_scan_type.ID AS mri_scan_type_ID FROM mri_protocol_checks LEFT JOIN mri_scan_type ON (mri_scan_type.ID=mri_protocol_checks.Scan_type) WHERE mri_scan_type.ID IS NULL AND mri_protocol_checks.Scan_type IS NOT NULL;

-- check that no SessionID are present in tarchive and not in session table
SELECT tarchive.SessionID AS tarchive_SessionID, session.ID AS session_ID, CandID AS session_CandID, Visit_label AS session_visit FROM tarchive LEFT JOIN session ON (session.ID=tarchive.SessionID) WHERE session.ID IS NULL AND tarchive.SessionID IS NOT NULL;
```

**Pull request merged**